tied-buffer/sycl-2.2/buffer-tags.md

```diff
@@ -54,14 +63,15 @@ class buffer {
 With this interface, a user can specify a specialisation of the buffer class by simply providing one or more values of the following tag enum class to the constructor of a buffer.
 
 ```cpp
-enum class tag {
-  context_bound,
-  mapping,
-  svm
+namespace property {
+  struct map_ptr;
+  struct cl_interop;
+  struct gl_interop;
+  struct svm;  // SYCL 2.2 Only
 };
 ```
 
-Some of these tags will require an additional interface for the buffer class, for example, the `tag::context_bound` tag will need a `get_context()` member function.
+Some of these tags will require an additional interface for the buffer class, for example, the `property::context_bound` tag will need a `get_context()` member function.
```

Comment:
If we are requiring that properties be passes as an instance, for example `property::context_bound(myContext)`, we could then potentially extend the `buffer` interface in a generic and extensible way through these objects. For example if you wanted to get the context of a context bound buffer you could do something like `buf.get_property<property::context_bound>().get_context()`.

This would remove any requirement to extend the `buffer` interface and handle exceptions for invalid uses of it.